app/components/global-search-box.js

```diff
@@ -1,50 +1,135 @@
 import Component from '@ember/component';
-import { task, timeout } from 'ember-concurrency';
+import { reads } from '@ember/object/computed';
+import { isBlank } from '@ember/utils';
 import { cleanQuery } from 'ilios-common/utils/query-utils';
+import { task, timeout } from 'ember-concurrency';
+import { addEventListener, runDisposables } from 'ember-lifeline';
 
 const DEBOUNCE_MS = 250;
 const MIN_INPUT = 3;
+
 export default Component.extend({
-  tagName: '',
+  classNames: ['global-search-box'],
+
+  'data-test-global-search-box': true,
+
+  initialQuery: null,
   query: null,
-  search(){},
+  savedQuery: null,
+  showResults: true,
+  onSearch() {},
+
+  isLoading: reads('autocomplete.isRunning'),
+  hasResults: reads('results.length'),
+  results: reads('autocomplete.lastSuccessful.value'),
+
+  init() {
+    this._super(...arguments);
+
+    if (this.initialQuery) {
+      this.set('query', this.initialQuery);
+    }
+  },
+
+  didRender() {
+    this._super(...arguments);
+    addEventListener(this, window, 'click', ({ target }) => {
+      if (!target.className.includes('no-action')) {
+        this.set('showResults', false);
+      }
+    });
+  },
 
   actions: {
+    focus() {
+      document.querySelector('input[type="search"]').focus();
+    },
+
     search() {
-      let q = cleanQuery(this.query);
+      const q = cleanQuery(this.query);
+
       if (q.length > 0) {
-        this.search(q);
+        this.onSearch(q);
       }
-    },
-    focus({ target }) {
-      //place focus into the search box when search icon is clicked
-      target.parentElement.parentElement.querySelector('input[type="search"]').focus();
-    },
+    }
   },
-  autocomplete: task(function* () {
-    let q = cleanQuery(this.query);
-    if (q.length === 0) {
-      return [];
+
+  destroy() {
+    runDisposables(this);
+    this._super(...arguments);
+  },
+
+  keyUp({ keyCode, target }) {
+    const fromInput = target.classList.contains('global-search-input');
+    const list = document.getElementsByClassName('autocomplete-row');
```

Comment:
The input would always have focus so the `target` element would also be the input. Any advice?

Reply:
You can ascend up from the `target` with `target.parentElement` since the DOM structure is part of this component it should be possible to go up and down it safely.

Reply:
👍